### PR TITLE
Updated Android samples for modern Android studio. Added OpenCV from Maven support.

### PR DESCRIPTION
**Pull request by alexlyulkov**
Updated samples for recent Android studio:

- added namespace field that is required in build.gradle files
- replaced _switch_ by _if-else_ because it doesn't work with constants from resources
- added missed log library dependency in face-detection/jni/CMakeLists.txt
- use local.properties to define NDK location

Added support for OpenCV from Maven. Now you can choose 3 possible sources of OpenCV lib in settings.gradle: SDK path, local Maven repository, public Maven repository. (Creating Maven repository from SDK is added here #24456 )

There are differences in project configs for SDK and Maven versions:

- different dependencies in build.gradle
- different OpenCV library names in CMakeLists.txt
- SDK version requires OpenCV_DIR definition

Requires:
- https://github.com/opencv/ci-gha-workflow/pull/124
- https://github.com/opencv-infrastructure/opencv-gha-dockerfile/pull/26

### CODE REVIEW

File: samples/android/15-puzzle/build.gradle.in [hunk 1]
```diff
@@ -27,5 +28,9 @@ android {
 
 dependencies {
     //implementation fileTree(dir: 'libs', include: ['*.jar'])
-    implementation project(':opencv')
+    if (gradle.opencv_source == "sdk_path") {
+        implementation project(':opencv')
+    } else if (gradle.opencv_source == "maven_local" || gradle.opencv_source == "maven_cenral") {
+        implementation 'org.opencv:opencv:@OPENCV_VERSION@'
```
Comment (asmorkalov) [inline samples/android/15-puzzle/build.gradle.in:34]: I propose to drop the version here and in all other samples.

**Review by asmorkalov — CHANGES_REQUESTED** [review]

File: cmake/android/android_gradle_projects.cmake [hunk 2]
```diff
@@ -1,3 +1,5 @@
+set (OPENCV_VERSION "4.8.0")
+
```
Comment (asmorkalov) [inline cmake/android/android_gradle_projects.cmake:2]: Not relevant.

**Comment by asmorkalov** [comment]
Namespace related: https://github.com/opencv/opencv/pull/23447

File: samples/android/face-detection/jni/CMakeLists.txt [hunk 3]
```diff
@@ -12,4 +17,5 @@ file(GLOB hdrs *.hpp *.h)
 
 include_directories("${CMAKE_CURRENT_LIST_DIR}")
 add_library(${target} SHARED ${srcs} ${hdrs})
-target_link_libraries(${target} ${ANDROID_OPENCV_COMPONENTS})
+find_library(log-lib log)
+target_link_libraries(${target} ${ANDROID_OPENCV_COMPONENTS} ${log-lib})
```
Comment (asmorkalov) [inline samples/android/face-detection/jni/CMakeLists.txt:21]: `log-lib` is bad name. let's use `log_lib`

File: samples/android/15-puzzle/build.gradle.in (hunk 1 above)
Comment (alexlyulkov) [reply samples/android/15-puzzle/build.gradle.in:34]: The version is required here

File: samples/android/camera-calibration/src/org/opencv/samples/cameracalibration/CameraCalibrationActivity.java [hunk 4]
```diff
@@ -136,23 +136,22 @@ public boolean onPrepareOptionsMenu (Menu menu) {
 
     @Override
     public boolean onOptionsItemSelected(MenuItem item) {
-        switch (item.getItemId()) {
```
Comment (asmorkalov) [inline samples/android/camera-calibration/src/org/opencv/samples/cameracalibration/CameraCalibrationActivity.java:139]: Related to: https://stackoverflow.com/questions/26753328/switch-case-with-r-id-xxxx-in-android-doesnt-work

**Review by asmorkalov — APPROVED** [review]
:+1:

File: .github/workflows/PR-4.x.yaml [hunk 5]
```diff
@@ -41,7 +41,7 @@ jobs:
     uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-iOS.yaml@main
 
   Android:
-    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-Android.yaml@main
+    uses: opencv/ci-gha-workflow/.github/workflows/OCV-PR-4.x-Android.yaml@as/android_tools_30
```
Comment (opencv-alalek) [inline .github/workflows/PR-4.x.yaml:44]: Need to revert that before merge.

**Review by opencv-alalek — APPROVED** [review]

**Pull request merged**